osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs

```diff
@@ -22,11 +23,28 @@ public class UserTotalPerformanceProcessor : IProcessor
         private BeatmapStore? beatmapStore;
         private BuildStore? buildStore;
 
+        // ReSharper disable once PrivateFieldCanBeConvertedToLocalVariable
+        private readonly Timer cacheResetTimer;
+
         private static readonly bool process_user_totals = Environment.GetEnvironmentVariable("PROCESS_USER_TOTALS") != "0";
 
         // This processor needs to run after the score's PP value has been processed.
         public int Order => ScorePerformanceProcessor.ORDER + 1;
 
+        public UserTotalPerformanceProcessor()
+        {
+            cacheResetTimer = new Timer(60000);
+            cacheResetTimer.Elapsed += (_, _) =>
+            {
+                // naive cache clearing, mainly to ensure new builds are picked up on new client releases.
+                beatmapStore = null;
+                buildStore = null;
```

Comment:
I think this will cause nullrefs because it's not locked with the usages.

Reply:
Is a very valid point.